154953781 Allow creation of users via API

kevinsamoei opened the pull request:
#### What does this PR do?
Allows users to be created via the REST API. This will help if third-party apps to be able to register user through their platforms easily.
#### Description of Task to be completed?
It should be possible to create new users via the API. External apps might need to register accounts on behalf of their users without having to send them over to the wger instance. A workflow like this is suggested:

Apps register as a regular user to get an API key
By default, users aren't allowed to create new accounts over the API, this has to be explicitly allowed
Users created with this key are marked somehow so it is later possible to identify them, you should probably think about adding a new OneToOne table to the user table "metadata" or such.

Simple idea for a basic protection against being spammed to death with new users:

REST API consumers that want to create users need a special right that has to be given on demand.
Users generated could have a special flag so that it's possible to see who created them

Note that it's probably easier and faster to add a new command for this (instead of a new element in the GUI). Something like `add-user-rest-api` and `list-user-rest-api `would be enough
#### How should this be manually tested?
The api can be accessed through this [URL](https://wg-logros.herokuapp.com/api/v2/users/) or append `/api/v2/users/` to access the user's endpoint. In order to give permissions to a user to be able to create a user, run the command `python manage.py add-user-rest-api --username <username>` and to list users created via the API run the command  `python manage.py list-user-rest-api`. 
#### Any background context you want to provide?
This was not possible previously and this PR provides the users of the API with a very powerful way to create users.
#### What are the relevant pivotal tracker stories?
[#154953781](https://www.pivotaltracker.com/story/show/154953781)

#### Screenshots (if appropriate)
* Authorized user adding a user
<img width="907" alt="screen shot 2018-02-28 at 18 35 56" src="https://user-images.githubusercontent.com/15800486/36796318-5a9d0fdc-1cb6-11e8-93cd-c2782e8b0002.png">
<img width="790" alt="screen shot 2018-02-28 at 18 36 16" src="https://user-images.githubusercontent.com/15800486/36796320-5ac8b65a-1cb6-11e8-96aa-a0c67157b833.png">
* Unauthorized user can not add a new user
<img width="782" alt="screen shot 2018-02-28 at 18 37 30" src="https://user-images.githubusercontent.com/15800486/36796388-81dfd9bc-1cb6-11e8-959f-db41730bd913.png">
* Authorize a user via the command line
<img width="645" alt="screen shot 2018-02-28 at 18 40 09" src="https://user-images.githubusercontent.com/15800486/36796553-d68805a2-1cb6-11e8-9e9f-f9c1ca3acb68.png">
<img width="649" alt="screen shot 2018-02-28 at 18 40 31" src="https://user-images.githubusercontent.com/15800486/36796575-e5ac4e26-1cb6-11e8-940b-a4eca2d61466.png">

* Listing users created via API
<img width="647" alt="screen shot 2018-02-28 at 18 41 26" src="https://user-images.githubusercontent.com/15800486/36796638-0c69f676-1cb7-11e8-9abb-7964804b8f0e.png">
<img width="643" alt="screen shot 2018-02-28 at 18 41 59" src="https://user-images.githubusercontent.com/15800486/36796677-26dda7fa-1cb7-11e8-93e2-9cdb11da63e1.png">

muchai-mercy commented on README.rst:
```diff
@@ -155,6 +155,36 @@ The available options for the ``wger`` command (if installed from PyPI) or
+To access the user endpoint got to:
+::
+
+http://localhost:8000/api/v2/users/
```
You want to avoid placing the exact local port. Just put a placeholder like http://*port_in_use*/api/v2/users

kevinsamoei replied:
Ok. Working on it.

kevinsamoei committed:
[Feature #154953781] Allow creation of users via REST API:
- Add user registration serializer
- Add user registration view
- Add create via api field to the models
- Add a url of the endpoint
- Add permissions for other users
- Add commands to create permissions and to list users via command line
- Add user endpoint information to the Readme

[Delivers #154953781]

muchai-mercy reviewed (approved):
LGTM
